commands/addons/destroy.js

```diff
@@ -12,6 +12,11 @@ function * run (context, heroku) {
   if (context.args.length === 0) throw new Error('Missing add-on name')
 
   let addons = yield context.args.map(name => resolve.addon(heroku, context.app, name))
+  for (let addon of addons) {
+    // prevent deletion of app when context.app is set but the addon is attached to a different app
+    let app = addon.app.name
+    if (context.app && app !== context.app) throw new Error(`${cli.color.addon(addons[0].name)} is on ${cli.color.app(app)} not ${cli.color.app(context.app)}`)
```

Comment:
@dickeyxxx I pushed up the change and am going to approve so that you can merge whenever